boot/boot_serial/src/boot_serial.c

```diff
@@ -468,25 +507,27 @@ bs_upload(char *buf, int len)
     }
 
 out:
+    if (fap != NULL) {
+        flash_area_close(fap);
+    }
     BOOT_LOG_INF("RX: 0x%x", rc);
     zcbor_map_start_encode(cbor_state, 10);
     zcbor_tstr_put_lit_cast(cbor_state, "rc");
     zcbor_uint32_put(cbor_state, rc);
     if (rc == 0) {
         zcbor_tstr_put_lit_cast(cbor_state, "off");
-        zcbor_uint32_put(cbor_state, curr_off);
+	zcbor_uint32_put(cbor_state, next_offset(curr_off, img_size));
     }
     zcbor_map_end_encode(cbor_state, 10);
 
-    boot_serial_output();
-    flash_area_close(fap);
-
 #ifdef MCUBOOT_ENC_IMAGES
-    if (curr_off == img_size) {
+    if (!img_upload_stalling && curr_off == img_size) {
```

Comment:
why to stall decryption at all?

Reply:
This skips decryption until stalling ends. If stalling is still happening then the image may not be something valid to decrypt yet.
There is also additional work needed in the future to actually add the decryption as "installation" step, so that the transfer reporting could also be stalled for the period it takes to decrypt an image.
Currently immediately when you get last offset of image the mcumgr-cli will get notified that upload has been successful, but what really happens is that decryption starts. This is actually wrong because you can no issue reset while in the middle of decrypting an image.